Allow suppression of console output

kivipe opened the pull request.

Atihinen commented:
Hi. Sorry for late response.

It seems that both `output` and `output_schema` has been modified but only `output` is tested. Personally I would mock out the `log_json` and check that it's called correctly with correct values. https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.assert_called_with

Also I would check that if it's written to console or not. e.g. https://stackoverflow.com/a/61533524/1271400

kivipe commented:
Thank you for your feedback and for pointing me in the right direction regarding mocks.

asimell commented on test/test_output.py:
```diff
@@ -0,0 +1,221 @@
+        }
+        self.output_console = '\n{\n    "robotframework": "cool"\n}\n'
+        self.output_schema_console = (
+            "\n"
+            + "{\n"
+            + '    "type": "object",\n'
+            + '    "properties": {\n'
+            + '        "robotframework": {\n'
+            '            "type": "string"\n'
+            "        }\n"
+            "    },\n"
+            + '    "required": [\n'
+            + '        "robotframework"\n'
+            + "    ]\n"
+            + "}\n"
```
This looks a bit weird with missing `+` and mixed quotes (`'` and `"`). Could you change this to a string block (`"""`)?

asimell replied:
Actually... Is this even needed?

Atihinen replied:
Tried this locally and removing the `self.output_schema_console` didn't affect to test outcome:
```
$ nox -s test -- test/test_output.py 
nox > Running session test
nox > Creating virtual environment (venv) using python3.8 in .venv/test
nox > python -m pip install --upgrade -r requirements-dev.txt
nox > pre-commit install
pre-commit installed at .git/hooks/pre-commit
nox > python -m unittest discover
....

Property 'abba' does not exist in:
{
    "schema": {
        "properties": "abba"
    }
}
.Unrecognized log level 'NOT ACCEPTABLE'. Using default log level 'WARN'.
.....................
----------------------------------------------------------------------
Ran 26 tests in 0.110s

OK
nox > pytest test/test_output.py
============================= test session starts ==============================
platform darwin -- Python 3.8.9, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
rootdir: /Users/jjauhiainen/projects/Eficode/oss/RESTKivipe
plugins: clarity-1.0.1, instafail-0.4.2, spec-3.2.0
collected 19 items                                                             

test/test_output.py ...................                                  [100%]

============================== 19 passed in 0.11s ==============================
nox > Session test was successful.
```

asimell commented:
@kivipe Starts to look good! One more comment from my side and then LGTM.

@Atihinen Any comments from you?

Atihinen commented:
@asimell  not from my side. Yeah to fix that not needed variable and we should be good to go with merge unless GHA finds something. 🙂

kivipe commented:
Thank you for your review. Variable self.output_schema_console was not used because somehow my output_schema tests were missing. I added output_schema tests into a new class so the purpose would be clearer. I rewrote init to avoid code duplication.

asimell reviewed (approved):
LGTM

Atihinen reviewed (approved):
Looks good to me
